staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -434,6 +439,18 @@ func NewCacherFromConfig(config Config) (*Cacher, error) {
 		)
 	}()
 
+	// Check to see if cacher supports RequestWatchProgress and warn if it does not
+	parentCtx := context.Background()
+	// Create a context with cancellation
+	ctx, cancel := context.WithCancel(parentCtx)
+	defer cancel()
+
+	// No need to do when storage supports the feature, just log error
+	_, err := cacher.Supports(ctx, storage.RequestWatchProgress)
+	if err != nil {
+		klog.ErrorS(err, "RequestWatchProgress feature does not support on this cluster")
+	}
```

Comment:
```suggestion
	if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) {
		supports, err := cacher.Supports(ctx, storage.RequestWatchProgress)
		if err != nil {
			klog.ErrorS(err, "Couldn't check features supported by etcd. Needed to use ConsistentListFromCache")
		}
		if !supports {
			klog.InfoS("RequestWatchProgress feature is not supported by provided etcd. This will prevent ConsistentListFromCache from working")
		}
	}
```